x/muxer/mapping.go

```diff
@@ -108,65 +108,54 @@ func (mapping mapping) layerSchema(dstSchema, srcSchema *schema.PackageSpec, src
 			srcSchema.Config.Required)
 	}
 
-	for k, v := range m.srcSchema.Config.Variables {
-		if m.dstSchema.Config.Variables == nil {
-			m.dstSchema.Config.Variables = map[string]schema.PropertySpec{}
-		}
-		// TODO: Validity check that this value matches any other config value
-		// with the same key
-		m.dstSchema.Config.Variables[k] = v
+	for k := range m.srcSchema.Config.Variables {
 		m.mapping.Config[k] = append(m.mapping.Config[k], m.srcIndex)
-		m.addType(v.TypeSpec)
 	}
+	layerMap(&m.dstSchema.Config.Variables, m.srcSchema.Config.Variables,
+		func(_ string, t schema.PropertySpec) { m.addType(t.TypeSpec) })
```

Comment:
We layer the config instead of setting it. In terms of our tests, this means we get SDKv2 config values instead of PF config values. SDKv2 values have defaults and descriptions.